Getting and Knowing your Data from ods.ai

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()

In [2]:
df = pd.read_csv(r'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv',  sep='\t')

In [3]:
df.head(10)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


In [5]:
df.shape

(4622, 5)

In [6]:
df.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

In [7]:
df.index

RangeIndex(start=0, stop=4622, step=1)

In [8]:
df['item_name'].value_counts()

Chicken Bowl                             726
Chicken Burrito                          553
Chips and Guacamole                      479
Steak Burrito                            368
Canned Soft Drink                        301
Steak Bowl                               211
Chips                                    211
Bottled Water                            162
Chicken Soft Tacos                       115
Chips and Fresh Tomato Salsa             110
Chicken Salad Bowl                       110
Canned Soda                              104
Side of Chips                            101
Veggie Burrito                            95
Barbacoa Burrito                          91
Veggie Bowl                               85
Carnitas Bowl                             68
Barbacoa Bowl                             66
Carnitas Burrito                          59
Steak Soft Tacos                          55
6 Pack Soft Drink                         54
Chips and Tomatillo Red Chili Salsa       48
Chicken Cr

In [9]:
df[['item_name', 'quantity']].groupby('item_name').sum().sort_values(['quantity'], ascending = False).head(1)

quantity
item_name             
Chicken Bowl       761

the most-ordered item is Chicken bowl with 726 orders and 761 items ordered

In [10]:
df['choice_description'].value_counts()

[Diet Coke]                                                                                                      134
[Coke]                                                                                                           123
[Sprite]                                                                                                          77
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese, Sour Cream, Lettuce]]                                            42
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese, Sour Cream, Guacamole, Lettuce]]                                 40
                                                                                                                ... 
[Fresh Tomato Salsa, [Black Beans, Pinto Beans, Cheese, Sour Cream, Lettuce]]                                      1
[Fresh Tomato Salsa, [Rice, Fajita Vegetables, Black Beans, Guacamole]]                                            1
[Tomatillo Green Chili Salsa, [Fajita Vegetables, Cheese, Sour C

In [11]:
df[['choice_description', 'quantity']].groupby('choice_description').sum().sort_values(['quantity'], ascending = False).head(1)

quantity
choice_description          
[Diet Coke]              159

the most-ordered choice description is Diet Coke with 134 orders and 159 items ordered

The price value of this dataset has the type of object. It needs to be switched to float

In [12]:
df.item_price.dtype

dtype('O')

In [13]:
df.item_price = df.item_price.apply(lambda x: float(x[1:-1]))

In [14]:
df.describe(include = 'all')

order_id     quantity     item_name choice_description   item_price
count   4622.000000  4622.000000          4622               3376  4622.000000
unique          NaN          NaN            50               1043          NaN
top             NaN          NaN  Chicken Bowl        [Diet Coke]          NaN
freq            NaN          NaN           726                134          NaN
mean     927.254868     1.075725           NaN                NaN     7.464336
std      528.890796     0.410186           NaN                NaN     4.245557
min        1.000000     1.000000           NaN                NaN     1.090000
25%      477.250000     1.000000           NaN                NaN     3.390000
50%      926.000000     1.000000           NaN                NaN     8.750000
75%     1393.000000     1.000000           NaN                NaN     9.250000
max     1834.000000    15.000000           NaN                NaN    44.250000

In [15]:
df.item_price.dtype

dtype('float64')

In [16]:
revenue = (df['item_price'] * df['quantity']).sum()
number_of_orders = df['order_id'].unique().max()
avg_revenue_per_order = revenue/number_of_orders
unique_items = df.item_name.value_counts().count()

In [17]:
print(
    ' the revenue for the period is', revenue, '$\n',
    'number of orders for the period is', number_of_orders, '\n',
    'the average revenue per order for the period is', '{0:.2f}'.format(avg_revenue_per_order), '\n',
    'total of', unique_items, 'unique items were sold'
)

 the revenue for the period is 39237.02 $
 number of orders for the period is 1834 
 the average revenue per order for the period is 21.39 
 total of 50 unique items were sold
